### PR TITLE
Added new sample, fixed bug, improved sample

### PR DESCRIPTION
**Pull request by astanley**
1) Ported swipemenu sample
2) fixed issue #2 for fileTransfer sample
3) improved verbiage within invoker sample

### CODE REVIEW

File: swipemenu/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+**Tested On**
+
+* BlackBerry 10 Dev Alpha 10.0.384
+* BlackBerry 10 Dev Alpha 10.0.386
+
+**Author(s)** 
+
+* [Adam Stanley](https://github.com/astanley)
+* [Rory Craig-Barnes](https://github.com/astanley)
```
Comment (cdelcol) [inline swipemenu/README.md]: Poor Rory. His Github profile link hijacked.

**Comment by kwallis** [comment]
Why are we adding the webworks.js file into the sample?

File: swipemenu/README.md (hunk 1 above)
Comment (astanley) [inline swipemenu/README.md]: haha, totally missed that.  I'll fixy fixy.

**Comment by astanley** [comment]
@kwallis - so it works when a developer downloads and hits compile.  All samples have the JS file in it.  Concerned about compatibility issues with future SDK updates?

**Comment by kwallis** [comment]
Yup.  Sorry, didn't realize the other samples had it as well.  Seems like a risky thing to do.  Better to have them figure it out and highlight the need to have the latest js file.  Only issue with that is that we don't have compatibility yet... hmm, still think that it would be better not to include and have explicit notes in the read me... Other thoughts?

## cc @nukulb

Ken Wallis

Product Manager – BlackBerry WebWorks

Research In Motion

(905) 629-4746 x14369

**Comment by astanley** [comment]
Alright, after some weird voodoo magic, I think I now have included Matt's change to the pullCapture sample in this pull request.  This change came from his pull request to the private repo, that I merged into my local copy, and then pushed up to this repo.  A bit confusing for me, but alas the change appears to be correct now.

**Comment by mdicesare** [comment]
Matt's changes look good to me

**Comment by nukulb** [comment]
@kwallis - I have been thinking about this as well.
I think we should remove the webworks.js and ask them to grab it from the SDK itself.

This file will change and cause problems in the future and these apps will have to upgraded with every release.

**Comment by mdandrea** [comment]
In our index.htm, we reference it not as webworks.js but as webworks-1.0.1.5.js so I guess you would want that to change to always just reference it as "webworks.js" and the user would have to rename their latest version to that?

**Comment by nukulb** [comment]
this might need a webworks.js naming change as well

**Comment by astanley** [comment]
That presents versioning issues. "Which webworks.js file do I have/need?"  Really think this file needs to maintain a version # as part of its filename to keep devs sane.

As Matt indicated, if we leave just the webworks-version.js file out of the project, the script reference tag(s) still exist in any HTML files.  If an app is going to fail, I'd rather it be the popup message that informs devs they need to upgrade the JS file, rather than runtime errors.

Suggestion: Keep it as is, but add an instruction to the README file about updating to the latest webworks-version.js file (where to find it, where to post it, how to update your HTML files, etc)?

**Comment by mdicesare** [comment]
Any update on the status of this pull request?

**Comment by nukulb** [comment]
@astanley - An interesting note for you, the dialog you see for incompatible version doesn't just use the version from the file name, it uses a build time hash of the webworks js file to pop up the dialog. This is done so that even if you have different versions in the file name but have the exact same file, it won't pop up the dialog. So if the version is removed from the webworks.js file it won't actually affect the dialog.

Also in my opinion we can completely remove all webworks.js  files from the this repo and tell the user to use from the SDK. So add a step in README telling the developer to grab it from the SDK before they package it. This will let our samples stay current otherwise we have to update them with every release. Also we will have to change the name (i.e remove the version from the name) for this to work.

Not saying its a great idea, simply putting it out there to see what people think

**Comment by astanley** [comment]
thanks for the info about the version prompt Nukul.  Good to know.

I'm game with making anything easier for developers. Agreed that the version number embedded in file names and HTML source makes it difficult to manage when we update the SDK.

I'll make an update to the pull request to remove the webworks.js files, update the README's and add a comment to the HTML files.

**Comment by mdicesare** [comment]
Adam, do you have an ETA on when these changes can be pulled in to the public repo? Our pull request is fairly urgent to go in since its preventing loading the WebWorks sample app for being signed because it was using a reserved keyword in the invoke target key.

**Comment by astanley** [comment]
Okay - have updated the files. Ready to push.  Will do so by EOD today unless there are any objections.

**Comment by mdandrea** [comment]
The changes look good to me.

-----Original Message-----

**Pull request merged**